File: src/main/java/guru/springframework/spring6webapp/service/BookService.java
package guru.springframework.spring6webapp.service;

import guru.springframework.spring6webapp.entity.Book;
import guru.springframework.spring6webapp.repository.BookRepository;
import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@AllArgsConstructor
public class BookService {
    private BookRepository bookRepository;

    public Iterable<Book> findAll() {
        return bookRepository.findAll();
    }

    public Book findById(Long id) {
        return bookRepository.findById(id).get();
    }

}

File: src/main/kotlin/guru/springframework/spring6webapp/service/BookService.kt
package guru.springframework.spring6webapp.service

import guru.springframework.spring6webapp.entity.Book
import guru.springframework.spring6webapp.repository.BookRepository
import org.springframework.stereotype.Service

@Service
class BookService(private val bookRepository: BookRepository) {
    fun findAll(): Iterable<Book> = bookRepository.findAll()

    fun findById(id: Long): Book = bookRepository.findById(id).get()
}
